### PR TITLE
dev/reference/sql: add sql statements

### PR DESCRIPTION
**Pull request by TomShawn**
Via: 
- https://github.com/pingcap/docs/pull/1067
- https://github.com/pingcap/docs-cn/pull/1400: merge with this PR

+ `ddl.md`, `dml.md`, `transaction.md`, `admin.md`, `prepare.md`, `util.md` have been replaced by 73 documents of sql statements.
+ There are 74 documents of sql statements in the [docs repository](https://github.com/pingcap/docs/tree/master/dev/reference/sql/statements) whereas the number in docs-cn repository are 73. The missing one is `admin.md`.

### CODE REVIEW

File: dev/reference/sql/statements/show-create-user.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+| Grants for root@%                         |
++-------------------------------------------+
+| GRANT ALL PRIVILEGES ON *.* TO 'root'@'%' |
++-------------------------------------------+
+1 row in set (0.00 sec)
+```
+
+## MySQL 兼容性
+
+* `SHOW CREATE USER` 的输出结果旨在匹配 MySQL，但 TiDB 尚不支持若干 `CREATE` 选项。尚未支持的选项将被解析但被忽略。详情可参阅 [security compatibility]。
```
Comment (dcalvin) [inline dev/reference/sql/statements/show-create-user.md]:
```suggestion
* `SHOW CREATE USER` 的输出结果旨在匹配 MySQL，但 TiDB 尚不支持若干 `CREATE` 选项。尚未支持的选项在语句执行过程中会被解析但会被跳过执行。详情可参阅 [security compatibility]。
```

File: dev/reference/sql/statements/deallocate.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
++------+
+1 row in set (0.00 sec)
+
+mysql> DEALLOCATE PREPARE mystmt;
+Query OK, 0 rows affected (0.00 sec)
+```
+
+## MySQL 兼容性
+
+`DEALLOCATE` 语句可视为与 MySQL 完全兼容。如有任何兼容性差异，请在 GitHub 上 提交 [issue](/report-issue.md)。
```
[the remark this answers is not in the document]
Comment (dcalvin) [reply dev/reference/sql/statements/deallocate.md]: 英文是 understood to be，感觉翻译为“可视为”有些语义上的模糊。如修改的话，请通篇修改。

File: dev/reference/sql/statements/delete.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+|  2 |  2 |
+|  3 |  3 |
+|  5 |  5 |
++----+----+
+4 rows in set (0.00 sec)
+```
+
+## MySQL 兼容性
+
+`DELETE` 语句可视为与 MySQL 完全兼容。如有任何兼容性差异，请在 GitHub 上 提交 [issue](/report-issue.md)。
```
[the remark this answers is not in the document]
Comment (dcalvin) [reply dev/reference/sql/statements/delete.md]: ditto

File: dev/reference/sql/statements/truncate.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
+---
+title: TRUNCATE
+summary: TiDB 数据库中 TRUNCATE 的使用概况。
+category: reference
+---
+
+# TRUNCATE
+
+`TRUNCATE` 语句以非事务方式从表中删除所有数据。可认为 `TRUNCATE` 语句同 `DROP TABLE` + `CREATE TABLE` 组合在语义上相同，定义与 `DROP TABLE` 语句相同。
```
[the remark this answers is not in the document]
Comment (TomShawn) [reply dev/reference/sql/statements/truncate.md]: `TRUNCATE` can be thought of as semantically the same as `DROP TABLE` + `CREATE TABLE` with the previous definition. “定义与 `DROP TABLE` ” 语句相同是指 "with the previous definition"。

File: dev/reference/sql/statements/commit.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+title: COMMIT
+summary: TiDB 数据库中 COMMIT 的使用概况。
+category: reference
+---
+
+# COMMIT
+
+`COMMIT` 语句用于在 TiDB 服务器内部提交事务。
+
+在没有 `BEGIN` 或 `START TRANSACTION` 语句的情况下，每个语句都为各自的事务和自动提交，这是 TiDB 的默认行为，确保了 MySQL 兼容性。
```
[the remark this answers is not in the document]
Comment (TomShawn) [reply dev/reference/sql/statements/commit.md]: Resolved: 在不使用 `BEGIN` 或 `START TRANSACTION` 语句的情况下，TiDB 中每一个查询语句本身也会默认作为事务处理，自动提交，确保了与 MySQL 的兼容。

File: dev/reference/sql/statements/begin.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+mysql> INSERT INTO t1 VALUES (1);
+Query OK, 1 row affected (0.00 sec)
+
+mysql> COMMIT;
+Query OK, 0 rows affected (0.01 sec)
+```
+
+## MySQL 兼容性
+
+`BEGIN` 语句与 MySQL 完全兼容。如有任何兼容性差异，请在 GitHub 上 [提交 issue](/report-issue.md)。
```
Comment (lilin90) [inline dev/reference/sql/statements/begin.md]:
```suggestion
`BEGIN` 语句与 MySQL 完全兼容。如有任何兼容性差异，请在 GitHub 上[提交 issue](/dev/report-issue.md)。
```

File: dev/reference/sql/statements/kill.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
++------+------+-----------+------+---------+------+-------+------------------+
+2 rows in set (0.00 sec)
+
+mysql> KILL TIDB 2;
+Query OK, 0 rows affected (0.00 sec)
+```
+
+## MySQL 兼容性
+
+* `KILL TIDB` 语句是 TiDB 的扩展语法。如果正尝试终止的会话位于同一个 TiDB 服务器上，可在配置文件里设置 [`compatible-kill-query = true`](/op-guide/tidb-config-file.md#compatible-kill-query)。
```
Comment (lilin90) [inline dev/reference/sql/statements/kill.md]:
```suggestion
* `KILL TIDB` 语句是 TiDB 的扩展语法。如果正尝试终止的会话位于同一个 TiDB 服务器上，可在配置文件里设置 [`compatible-kill-query = true`](/dev/reference/configuration/tidb-server/configuration-file.md#compatible-kill-query)。
```

**Pull request merged**